sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformationExec.scala

```diff
@@ -146,12 +148,32 @@ case class ScriptTransformationExec(
           }
         }
 
+        private def waitForTransformationProcessTermination(): Unit = {
+          val timeout = SQLConf.get.getConf(SQLConf.TRANSFORMATION_EXIT_TIMEOUT)
+          try {
+            if (timeout < 0) {
+              proc.waitFor()
+            } else if (timeout == 0) {
+              // Do nothing
+            } else {
+              proc.waitFor(timeout, TimeUnit.SECONDS)
+            }
```

Comment:
I feel that the logical here is too complex. It really does not make sense to allow user to set 0 timeout which could introduce the potential bug again. Also negative timeout may cause app hang, which breaks our initial motivation of avoiding app hang.

Thus, I think we should only allow timeout > 0 and add `checkValue(timeout > 0)` to the conf and simplify the code here.

Reply:
+1

Reply:
Sure,  updated.